infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -265,171 +250,252 @@ public static void main(String[] args) {
+
+        /** Configure a {@link ServerBootstrap}.
+         *
+         * @param serverBootstrap   The {@link ServerBootstrap} to configure.
+         */
+        void configure(ServerBootstrap serverBootstrap);
+    }
+
+    /** Start the Corfu server and bind it to the given {@code port} using the provided
+     * {@code channelType}. It is the callers' responsibility to shutdown the
+     * {@link EventLoopGroup}s. For implementations which listen on multiple ports,
+     * {@link EventLoopGroup}s may be reused.
+     *
+     * @param bossGroup             The "boss" {@link EventLoopGroup} which services incoming
+     *                              connections.
+     * @param workerGroup           The "worker" {@link EventLoopGroup} which services incoming
+     *                              requests.
+     * @param channelType           The type of {@link ServerChannel} to use.
+     * @param bootstrapConfigurer   A {@link BootstrapConfigurer} which will receive the
+     *                              {@link ServerBootstrap} to set options.
+     * @param context               A {@link ServerContext} which will be used to configure the
+     *                              server.
+     * @param router                A {@link NettyServerRouter} which will process incoming
+     *                              messages.
+     * @param port                  The port the {@link ServerChannel} will be created on.
+     * @param <T>                   The type of the {@link ServerChannel}.
+     * @return                      A {@link ChannelFuture} which can be used to wait for the server
+     *                              to be shutdown.
+     */
+    public static <T extends ServerChannel>
+            ChannelFuture startAndListen(@Nonnull EventLoopGroup bossGroup,
+                          @Nonnull EventLoopGroup workerGroup,
+                          @Nonnull Class<T> channelType,
+                          @Nonnull BootstrapConfigurer bootstrapConfigurer,
+                          @Nonnull ServerContext context,
+                          @Nonnull NettyServerRouter router,
+                          int port) {
+        try {
+            ServerBootstrap bootstrap = new ServerBootstrap();
+            bootstrap.group(bossGroup, workerGroup)
+                .channel(channelType);
+            bootstrapConfigurer.configure(bootstrap);
+
+            bootstrap.childHandler(getServerChannelInitializer(context, router));
+            return bootstrap.bind(port).sync();
+        } catch (InterruptedException ie) {
+            throw new UnrecoverableCorfuInterruptedError(ie);
+        }
+    }
+
+    /** Configure server bootstrap per-channel options, such as TCP options, etc.
+     *
+     * @param context       The {@link ServerContext} to use.
+     * @param bootstrap     The {@link ServerBootstrap} to be configured.
+     */
+    public static void configureBootstrapOptions(@Nonnull ServerContext context,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "context". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)